src/Main.enso

```diff
@@ -1,105 +1,193 @@
+product x=1 y=2 = x * y
 
-    # Operators have different precedence depending on spacing
+# They can then be called with default arguments by name
+result3 = product x=20 y=100 # equivalent to product 20 100
+result4 = product x=20 # equivalent to product 20 2
+result5 = product 20 # equivalent to product 20 2
+result6 = product 20 ... # returns a function expecting a parameter for y
+
+# The entry point for an Enso program is a method `main` in `Main.enso`.
+main =
+    # Methods can be called with `.`.
+    # Methods can be passed additional arguments.
+    # `up_to` creates a `Range` from 1 up to 10.
+    range = 1.up_to 10
+
+    # Operators, including `.`, have different precedence depending on spacing
     # No-spaces binds stronger than a spaced version
     # foo bar . baz ==> (foo bar).baz
     # foo bar.baz   ==> foo (bar.baz)
-    operator1 = 0.up_to 100 . to_vector . map .noise
+    sorted_list = range . to_vector . sort Ascending
+    # equivalent to:
+    # sorted_list = (range.to_vector).sort Ascending
 
-    # Operator sections!
+    # Operator sections:
     # (+ 5) => a function that takes x and returns x+5
-    # (5 +) => ...
+    # (5 +) => as above.
     # [1, 2, 3] . map (+ 5)
     # [1, 2, 3] . map +5
     # .noise => a function that takes x and return x.noise
-    operator1.sort
+    sorted_list.sort  # a function expecting a comparator to sort on.
 
-    # Lambdas
+    # Lambdas / anonymous functions
     # Created with `->`
-    # Arguments go on the left, result goes on the right
+    # Parameters on the left, results on the right:
     [1, 2, 3] . map (x -> x * 5)
+    # Multiple parameters are achieved by multiple `->`:
     [1, 2, 3] . map z-> y-> x->
         y_1 = x * 5
         z_1 = y_1 + 3
         z_1
     [1, 2, 3] . reduce (x -> y -> x + y)
 
     # Lambda shorthands
-    my_lambda = (here.my_method _ 5) # equivalent to (x-> here.my_method x 5)
-    my_lambda_2 = (here.my_method _ 6 _ 5) # equivalent to (y-> x-> here.my_method y 6 x 5)
+    my_lambda = (here.sum _ 5) # equivalent to (x-> here.sum x 5)
 
     # Calling a method
-    this.my_method 1 2 # `this` refers to whatever the current method is called on
-    here.my_method 1 2 # `here` ALWAYS refers to the current module
+    this.sum 1 2 # `this` refers to whatever the current method is called on
+    here.sum 1 2 # `here` ALWAYS refers to the current module
 
-    # Calling a method with default arguments, by name
-    my_circ_1 = here.draw_circle x_position=37 radius=7 # equivalent to here.draw_circle 7 'red' 37 0
-    my_circ_2 = here.draw_circle 36 'blue' # equivalent to here.draw_circle 36 'blue' 0 0
-    my_circ_factory = here.draw_circle 36 'blue' ... # equivalent to (x=0)-> (y=0)-> here.draw_circle 36 'blue' x y
-
-my_method arg1 arg2 =
-    IO.println arg1
-    IO.println arg2
     # The last line in a block of code is the return value
-    arg1 + arg2
+    IO.println here.sum 1 99
```

Comment:
```suggestion
    IO.println (here.sum 1 99)
```

Also this would probably be better without the `println` – print returns `Nothing` so the return value is very uninteresting.